compiler/rustc_lint/src/types.rs

```diff
@@ -727,7 +730,33 @@ declare_lint! {
     "proper use of libc types in foreign item definitions"
 }
 
-declare_lint_pass!(ImproperCTypesDefinitions => [IMPROPER_CTYPES_DEFINITIONS]);
+declare_lint! {
+    /// The AIX target follows the power alignment rule. This rule specifies
+    /// that structs with either a:
+    ///   - floating-point data type as its first member, or
+    ///   - first member being an aggregate whose recursively first member is a
+    ///     floating-point data type
+    /// must have its natural alignment. This is currently unimplemented within
```

Comment:
"natural alignment"?

What does "natural alignment" mean?

The alignment of f64 is 8 bytes, so the natural alignment of the struct should be 8 bytes, and the produced struct will be 8 bytes, but the [alignment to 4 bytes described here](https://internals.rust-lang.org/t/repr-c-aix-struct-alignment/21594) by @mustartt is not a natural alignment.